File: examfront/src/app/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private http: HttpClient) { }

  public allQuizzes(){
    return this.http.get(`${baseUrl}/quiz/`);
  }


  //add quiz
  public addQuiz(quiz: any){

    return this.http.post(`${baseUrl}/quiz/`,quiz);
  }


  //delete quiz
  public deleteQuiz(qid: any)
  {
    return this.http.delete(`${baseUrl}/quiz/${qid}`);
  }


  //get the single quiz
  public getQuiz(qid: any){
    return this.http.get(`${baseUrl}/quiz/${qid}`);
  }

  //Update Quiz
  public updateQuiz(quiz: any){
    return this.http.put(`${baseUrl}/quiz/`,quiz);
  }

  //get quizes of category
  public getQuizzesOfCategory(cid:any){
    return this.http.get(`${baseUrl}/quiz/category/${cid}`);
  }


 //get active quizzes
 public getActiveQuizzes(){
   return this.http.get(`${baseUrl}/quiz/active`);
 }


 //get active quizzes of Category
 public getActiveQuizzesOfCategory(cid:any){
   return this.http.get(`${baseUrl}/quiz/category/active/${cid}`);
 }
}

File: examserver/src/main/java/com/exam/services/QuizService.java
package com.exam.services;

import com.exam.model.exam.Category;
import com.exam.model.exam.Quiz;
import org.springframework.http.ResponseEntity;

import java.util.List;
import java.util.Set;

public interface QuizService {

     Quiz addQuiz(Quiz quiz);
     Quiz updateQuiz(Quiz quiz);
     Set<Quiz> getQuizzes();
     Quiz getQuiz(Long quizId);
     void deleteQuiz(Long quizId);

    List<Quiz> getQuizOfCategory(Category c);

    List<Quiz> getActiveQuizzes();
    List<Quiz> getActiveQuizzesOfCategory(Category c);
}
